cli/fathom_web/test/test_list.py

```diff
@@ -0,0 +1,176 @@
+from click.testing import CliRunner
+
+from ..commands.list import main as list_main
+
+
+def test_end_to_end(tmp_path):
+    """Test expected outcome when using all of the optional parameters"""
+    # Make temporary in_directory and base_dir directories
+    base_dir, in_directory = make_directories(tmp_path)
+
+    # Make HTML files in in_directory in two separate subdirectories
+    # so we can exercise the recursive option.
+    a1, a2, b1, b2 = make_html_files(in_directory)
+
+    # Make the out_file we will save the output to
+    out_file = (base_dir / 'out_file.txt')
+
+    # Run fathom-list
+    result = CliRunner().invoke(
+        list_main,
+        [
+            in_directory.as_posix(),
+            f'-b{base_dir.as_posix()}',
+            '-r',
+            f'-o{out_file.as_posix()}',
+        ],
+    )
+    assert result.exit_code == 0
+
+    expected_file_contents = [
+        a1.relative_to(base_dir).as_posix(),
+        a2.relative_to(base_dir).as_posix(),
+        b1.relative_to(base_dir).as_posix(),
+        b2.relative_to(base_dir).as_posix(),
+    ]
+    actual_file_contents = out_file.read_text().splitlines()
+    assert expected_file_contents == actual_file_contents
```

Comment:
This test is failing for me (see [verbose pytest output](https://paste.mozilla.org/NLTQUNU0)).

```
cli/fathom_web/test/test_list.py:39: AssertionError
------------------------------- Captured stdout call --------------------------------
actual_file_contents: $['in_directory/source_a/2.html', 'in_directory/source_a/1.html', 'in_directory/source_b/2.html', 'in_directory/source_b/1.html']
expected_file_contents: $['in_directory/source_a/1.html', 'in_directory/source_a/2.html', 'in_directory/source_b/1.html', 'in_directory/source_b/2.html']
```

It looks like checking two lists for equality in Python requires the list elements to have the same values in the same order, but in my case, the values are the same but the order is different -- though I'm not sure why this doesn't happen for you or for the Travis machine. Since order doesn't matter in this case, can we check for the same values in any order instead?
I have to imagine perhaps the way my OS does recursive search could be different?

Reply:
Very interesting. I'll switch it to a set since that's conceptual more appropriate.